dex-it/src/test/scala/com/wavesplatform/it/sync/OrderFeeTestSuite.scala

```diff
@@ -0,0 +1,430 @@
+         |    }
+         |  }
+         |}
+       """.stripMargin
+
+    super.nodeConfigs.map(
+      ConfigFactory
+        .parseString(orderFeeSettingsStr)
+        .withFallback
+    )
+  }
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    val txIds = Seq(IssueUsdTx, IssueEthTx, IssueBtcTx).map(_.json()).map(node.broadcastRequest(_).id)
+    txIds.foreach(node.waitForTransaction(_))
+  }
+
+  "supported non-waves order fee" - {
+    val btcRate = 0.0005
+    val ethRate = 0.0064
+
+    "is not enough" in {
+      Map(BtcId -> btcRate, EthId -> ethRate)
+        .foreach(asset => node.upsertRate(IssuedAsset(asset._1), asset._2, expectedStatusCode = StatusCodes.Created))
+
+      assertBadRequestAndResponse(
+        node.placeOrder(
+          node.prepareOrder(
+            sender = bob,
+            pair = AssetPair(Waves, IssuedAsset(BtcId)),
+            orderType = OrderType.BUY,
+            amount = 1.waves,
+            price = 50000L,
+            fee = 100L,
+            version = 3: Byte,
+            matcherFeeAssetId = IssuedAsset(BtcId))
+        ), s"Required 0.0000015 $BtcId as fee for this order, but given 0.000001 $BtcId")
+
+      assertBadRequestAndResponse(
+        node.placeOrder(
+          node.prepareOrder(
+            sender = bob,
+            pair = AssetPair(Waves, IssuedAsset(BtcId)),
+            orderType = OrderType.BUY,
+            amount = 1.waves,
+            price = 50000L,
+            fee = 1920L,
+            version = 3: Byte,
+            matcherFeeAssetId = IssuedAsset(EthId))
+        ), s"Not enough tradable balance. The order requires 0.0000192 $EthId and 0.0005 $BtcId"
+      )
+      Array(BtcId, EthId)
+        .foreach(assetId => node.deleteRate(IssuedAsset(assetId)))
+    }
+
+    "is enough" in {
+      node.upsertRate(IssuedAsset(BtcId), btcRate, expectedStatusCode = StatusCodes.Created)
+      node.placeOrder(
+        node.prepareOrder(
+          sender = bob,
+          pair = AssetPair(Waves, IssuedAsset(BtcId)),
+          orderType = OrderType.SELL,
+          amount = 1.waves,
+          price = 50000L,
+          fee = 150L,
+          version = 3: Byte,
+          matcherFeeAssetId = IssuedAsset(BtcId)
+        )
+      )
+      node.reservedBalance(bob).keys shouldNot contain (BtcId.toString)
+      node.reservedBalance(bob)("WAVES") shouldEqual 100000000L
+      node.cancelAllOrders(bob)
+
+      node.placeOrder(
+        node.prepareOrder(
+          sender = bob,
+          pair = AssetPair(Waves, IssuedAsset(BtcId)),
+          orderType = OrderType.BUY,
+          amount = 1.waves,
+          price = 50000L,
+          fee = 150L,
+          version = 3: Byte,
+          matcherFeeAssetId = IssuedAsset(BtcId)
+        )
+      )
+      node.reservedBalance(bob)(BtcId.toString) shouldEqual 50150L
+      node.reservedBalance(bob).keys shouldNot contain("WAVES")
+      node.cancelAllOrders(bob)
+      node.deleteRate(IssuedAsset(BtcId))
```

Comment:
add: bob has some amount for fee, but not enough to pay fee, additional fee should be withdraw after order fill